plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -812,12 +816,15 @@ private static ReaderPageSource createParquetPageSource(
             long nextStart = 0;
             ImmutableList.Builder<Long> blockStarts = ImmutableList.builder();
             List<BlockMetaData> blocks = new ArrayList<>();
+            ImmutableList.Builder<Optional<ColumnIndexStore>> columnIndexes = ImmutableList.builder();
             for (BlockMetaData block : parquetMetadata.getBlocks()) {
                 long firstDataPage = block.getColumns().get(0).getFirstDataPageOffset();
+                Optional<ColumnIndexStore> columnIndex = getColumnIndexStore(dataSource, block, descriptorsByPath, parquetTupleDomain, options);
```

Comment:
Although we won't read the column index from file until later, it is better to avoid creating `columnIndex` until it's needed (after the `start <= firstDataPage && firstDataPage < start + length`). We can make same change in ParquetPageSourceFactory as well.